android/titanium/src/java/org/appcelerator/titanium/io/TiFile.java

```diff
@@ -113,6 +113,17 @@ public boolean createDirectory(boolean recursive)
 		}
 	}
 
+    @Override
+    public boolean createFile() {
```

Comment:
I don't think this will compile. You're trying to re-throw the IOException and the method doesn't declare it throws that exception type (and neither does the super implementation we're overriding).

Also, if you're just catching an exception and rethrowing it, you don't really need to do that, you're basically just doing what would have already happened anyways.

So you either need to declare here and in TiBaseFile that #createFile() throws IOException, or you need: catch it, ignore or log it, and return false.

Given that our API doesn't mention throwing errors on failure, but uses a straight boolean, I'd keep the method signature as is and do:

```java
@Override
public boolean createFile() {
	try {
		return file.createNewFile();
	} catch (IOException e) {
		Log.e(TAG, "Error creating new file: ", e);
		return false;
	}
}
```